components/log-ingestor/tests/test_ingestion_job.rs

```diff
@@ -140,25 +178,110 @@ async fn test_sqs_listener() -> Result<()> {
+    let s3_client = clp_rust_utils::s3::create_new_client(
+        aws_config.endpoint.as_str(),
+        aws_config.region.as_str(),
+        aws_config.access_key_id.as_str(),
+        &secret_access_key,
     )
-    .await
-    .expect("Timed out while receiving object metadata");
+    .await;
+
+    let upload_and_receive_handle = tokio::spawn(upload_and_receive(
+        s3_client.clone(),
+        aws_config.bucket_name.clone(),
+        prefix.clone(),
+        NUM_TEST_OBJECTS,
+        receiver,
+    ));
+    let noise_upload_handle = tokio::spawn(upload_noise_objects(
+        s3_client.clone(),
+        aws_config.bucket_name.clone(),
+        NUM_TEST_OBJECTS,
+    ));
+
+    noise_upload_handle
+        .await
+        .expect("Error while awaiting noise upload");
+    let (mut created_objects, mut received_objects) = upload_and_receive_handle
+        .await
+        .expect("Error while awaiting upload and receive");
+
     sqs_listener.shutdown_and_join().await?;
 
-    objects_to_create.sort();
+    created_objects.sort();
+    received_objects.sort();
+    assert_eq!(received_objects, created_objects);
+
+    Ok(())
+}
+
+#[tokio::test]
+#[serial_test::serial]
+#[ignore = "Requires LocalStack or AWS environment"]
+async fn test_s3_scanner() -> Result<()> {
+    let job_id = Uuid::new_v4();
+    let prefix = get_testing_prefix(&job_id);
+
+    let aws_config = AwsConfig::from_env()?;
+
+    let secret_access_key =
+        secrecy::SecretString::new(aws_config.secret_access_key.clone().into_boxed_str());
+    let s3_client = clp_rust_utils::s3::create_new_client(
+        aws_config.endpoint.as_str(),
+        aws_config.region.as_str(),
+        aws_config.access_key_id.as_str(),
+        &secret_access_key,
+    )
+    .await;
+
+    let s3_scanner_config = S3ScannerConfig {
+        bucket_name: aws_config.bucket_name.clone(),
+        prefix: prefix.clone(),
+        scanning_interval: Duration::from_millis(300),
+        start_after: None,
+    };
+
+    let (sender, receiver) = mpsc::channel::<ObjectMetadata>(TEST_CHANNEL_CAPACITY);
+
+    let s3_scanner = log_ingestor::ingestion_job::S3Scanner::spawn(
+        job_id,
+        S3ClientWrapper::from(s3_client.clone()),
+        s3_scanner_config,
+        sender,
+    );
+
+    let upload_and_receive_handle = tokio::spawn(upload_and_receive(
+        s3_client.clone(),
+        aws_config.bucket_name.clone(),
+        prefix.clone(),
+        NUM_TEST_OBJECTS,
+        receiver,
+    ));
+    let noise_upload_handle = tokio::spawn(upload_noise_objects(
+        s3_client.clone(),
+        aws_config.bucket_name.clone(),
+        NUM_TEST_OBJECTS,
+    ));
+
+    noise_upload_handle
+        .await
+        .expect("Error while awaiting noise upload");
+    let (mut created_objects, mut received_objects) = upload_and_receive_handle
+        .await
+        .expect("Error while awaiting upload and receive");
+    s3_scanner.shutdown_and_join().await?;
```

Comment:
Sorry I didn't notice this last time. I guess even in test function, we should either use `anyhow::Context` or `Result::expect`. It makes little sense to mixing unwrap/expect with Result in test. (We need to update those upload functions if we choose the Result route)
either 

```suggestion
    noise_upload_handle
        .await
        .context("Error while awaiting noise upload")?;
    let (mut created_objects, mut received_objects) = upload_and_receive_handle
        .await
        .context("Error while awaiting upload and receive")?;
    s3_scanner.shutdown_and_join().await?;
```

or
```suggestion
    noise_upload_handle
        .await
        .expect("Error while awaiting noise upload");
    let (mut created_objects, mut received_objects) = upload_and_receive_handle
        .await
        .expect("Error while awaiting upload and receive");
    s3_scanner.shutdown_and_join().await.unwrap();
```